contents/staticAnalysis/PMD.md

```diff
@@ -0,0 +1,86 @@
+## PMD
+PMD is a static analyser for Java, JavaScript, Salesforce.com Apex, PL/SQL, Apache Velocity, XML, XSL. The copy/paste-detector([CPD](http://pmd.sourceforge.net/pmd-4.3.0/cpd.html)), which helps to find duplicated code, is also included as an add-on in PMD.
+
+### Features
+Rules in PMD represent patterns in code. PMD is supposed to check these rules(patterns) and signal violations to programmers. For example, the `OverrideBothEqualsAndHashcode` rule requires programmers to override both `public boolean Object.equals(Object other)`, and `public int Object.hashCode()`, or override neither. 
+
+PMD supports checking rules for the following languages.
+
+- [Java](https://pmd.github.io/pmd-5.5.3/pmd-java/index.html) (includes [JSP](https://pmd.github.io/pmd-5.5.3/pmd-java/index.html) with [limitations](https://pmd.github.io/pmd-5.5.3/pmd-jsp/index.html))
+- [JavaScript](https://pmd.github.io/pmd-5.5.3/pmd-javascript/index.html)
+- [Apex](https://pmd.github.io/pmd-5.5.3/pmd-apex/index.html)
+- [PL/SQL](https://pmd.github.io/pmd-5.5.3/pmd-plsql/index.html)
+- [Velocity](https://pmd.github.io/pmd-5.4.1/pmd-vm/index.html)
+- [XML and XSL](https://pmd.github.io/pmd-5.5.3/pmd-xml/index.html)
+
+PMD doesn't support checking rules for the following languages. Only Copy/Paste Detector ([CPD](http://pmd.sourceforge.net/pmd-4.3.0/cpd.html)) is supported for them.
+
+- [C++](https://pmd.github.io/pmd-5.5.3/pmd-cs/index.html)
+- [C#](https://pmd.github.io/pmd-5.5.3/pmd-cpp/index.html)
+- [Fortran](https://pmd.github.io/pmd-5.5.3/pmd-fortran/index.html)
+- [Go](https://pmd.github.io/pmd-5.5.3/pmd-go/index.html)
+- [Groovy](https://pmd.github.io/pmd-5.5.3/pmd-groovy/index.html)
+- [Matlab](https://pmd.github.io/pmd-5.5.3/pmd-matlab/index.html)
+- [Objective-C](https://pmd.github.io/pmd-5.5.3/pmd-objectivec/index.html)
+- [Perl](https://pmd.github.io/pmd-5.5.3/pmd-perl/index.html) (only very basic CPD support)
+- [PHP](https://pmd.github.io/pmd-5.5.3/pmd-php/index.html)
+- [Python](https://pmd.github.io/pmd-5.5.3/pmd-python/index.html)
+- [Ruby](https://pmd.github.io/pmd-5.5.3/pmd-ruby/index.html)
+- [Swift](https://pmd.github.io/pmd-5.5.3/pmd-swift/index.html)
+- [Scala](https://pmd.github.io/pmd-5.5.3/pmd-scala/index.html)
```

Comment:
Where possible, It is better to refer to a stable/official external documentation instead of duplicating the info in our docs. Reason: our docs can get outdated soon.